packages/origin-ui-core/src/components/Modal/OrderDetailslModal.tsx

```diff
@@ -0,0 +1,199 @@
+import React from 'react';
+import { Dialog, Grid, IconButton, useTheme, Box, Divider, Button } from '@material-ui/core';
+import { Order } from '../../utils/exchange';
+import { Close as CloseIcon, ArrowRightAlt as ArrowRightAltIcon } from '@material-ui/icons';
+import {
+    moment,
+    formatCurrencyComplete,
+    getCurrencies,
+    getEnvironment,
+    getProducingDevices
+} from '../..';
+import { useSelector } from 'react-redux';
+import { OrderSide } from '@energyweb/exchange-core';
+import { EnergyFormatter, deviceById, useTranslation } from '../../utils';
+
+interface IOwnProps {
+    order: Order;
+    close: () => void;
+    showCancelOrder: (order: Order) => void;
+}
+
+export const OrderDetailsModal = (props: IOwnProps) => {
+    const { t } = useTranslation();
+    const anyOption = t('order.anyValue');
+    const { order, close, showCancelOrder } = props;
+    const {
+        filled,
+        product: {
+            generationFrom,
+            generationTo,
+            location,
+            deviceType,
+            gridOperator = anyOption,
+            externalDeviceId
+        }
+    } = order;
+    const {
+        spacing,
+        palette: {
+            text: { primary: textPrimary }
+        },
+        typography: { fontSizeSm, fontSizeLg }
+    } = useTheme();
+    const currency = useSelector(getCurrencies)[0];
+    const primaryDeviceType = deviceType ? deviceType[0].split(';')[0] : anyOption;
+    const region = location ? location[0].split(';')[1] : anyOption;
+    const devices = useSelector(getProducingDevices);
+    const environment = useSelector(getEnvironment);
+
+    return (
+        <Dialog
+            open={order !== null}
+            onClose={close}
+            style={{ fontSize: fontSizeSm }}
+            className="OrderModal"
+            fullWidth={true}
+        >
+            <Grid container direction="column" style={{ alignItems: 'stretch' }}>
+                <Grid
+                    item
+                    container
+                    direction="column"
+                    style={{ paddingLeft: spacing(2), paddingRight: spacing(2) }}
+                >
+                    <Grid item style={{ alignSelf: 'end' }}>
+                        <IconButton onClick={close}>
+                            <CloseIcon style={{ color: textPrimary }} />
+                        </IconButton>
+                    </Grid>
+                    <Grid item style={{ alignSelf: 'start' }}>
+                        <Box pb={1} fontWeight="fontWeightLight">
+                            {moment(order.validFrom).format('MMM Do, YYYY h:mm:ss a')}
```

Comment:
We already have the default time format defined here: https://github.com/energywebfoundation/origin/blob/2a6ea431bde2092d545c3ad0e32016b8a81a05e6/packages/origin-ui-core/src/utils/time.ts#L5